torch_glow/src/GlowFuser.cpp

```diff
@@ -51,140 +49,274 @@ sortReverseTopological(at::ArrayRef<torch::jit::Value *> inputs,
   return result;
 }
 
-bool canMerge(torch::jit::Node *node, isSupportFunc fn) {
-  return node->kind() == torch::jit::prim::Constant || fn(node);
-}
+bool canMerge(torch::jit::Node *node, IsSupportFunc fn,
+              torch::jit::Node *consumer) {
+  if (node->kind() == torch::jit::prim::Param) {
+    return false;
+  }
 
-#define REQ(cond, ignore, log_info)                                            \
-  if (!(cond)) {                                                               \
-    if (!(ignore)) {                                                           \
-      DLOG(ERROR) << (log_info);                                               \
-    }                                                                          \
-    return c10::nullopt;                                                       \
+  // Check that the node is supported
+  if (!(fn(node) || node->kind() == torch::jit::prim::Constant ||
```

Comment:
We should merge the kind() to the first if?

Reply:
I prefer this way because the first if is a special case we explicitly don't support, this if is the opposite, these are nodes we do support including some special cases